src/App.tsx

```diff
@@ -1,26 +1,310 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState,
+  useMemo,
+  useEffect,
+  useRef,
+  useCallback,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodos,
+  TodosErrorsService,
+  todosErrorsServiceText,
+  todosService,
+  USER_ID,
+} from './api/todos';
+import { getFilteredTodos, Todo } from './types/Todo';
+import cn from 'classnames';
+import { TodoItems } from './components/TodoItems';
+import {
+  TodoStatusFilter,
+  TODO_STATUS_FILTER_OPTIONS,
+} from './components/TodoStatusFilter';
+import { useErrorMessage } from './components/hooks/useErrorMessage';
+import { TodoCreate } from './types/TodoCreate';
+import { CreateTodoForm } from './components/CreateTodoForm';
+import { useLoadingTodos } from './components/hooks/useLoadingTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  // const [loadingTodoIds, setLoadingTodoIds] = useState<Todo['id'][]>([]);
+  const [selectedStatus, setSelectedStatus] = useState(TodoStatusFilter.All);
+  const [isLoadingTodos, setIsLoadingTodos] = useState(true);
+  const { errorMessage, handleRemoveError, handleSetError } = useErrorMessage();
+  const { loadingTodoIds, handleAddTodoToLoading, handleRemoveTodoToLoading } =
+    useLoadingTodos();
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const complitedTodos = todos.filter(todo => todo.completed);
+  const todoTitleInputRef = useRef<HTMLInputElement>(null);
+
+  const handleDeleteTodo = (todoId: Todo['id']) => {
+    handleAddTodoToLoading(todoId);
+    handleRemoveError();
+    todosService
+      .deleteTodods(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        handleSetError(
+          todosErrorsServiceText[TodosErrorsService.UNABLE_TO_DELETE_TODO],
+        );
+      })
+      .finally(() => {
+        handleRemoveTodoToLoading(todoId);
+        todoTitleInputRef.current?.focus();
+      });
+  };
+
+  const handleDeleteBulk = (todoId: Todo['id'][]) => {
+    todoId.forEach(id => handleDeleteTodo(id));
+  };
+
+  const handleDeleteComplited = () => {
```

Comment:
Here and in multiple other places

```suggestion
  const handleDeleteCompleted = () => {
```